vmdk_plugin/drivers/vmdk/vmdk_driver.go

```diff
@@ -102,25 +102,25 @@ func (d *VolumeDriver) VolumesInRefMap() []string {
 // map hasn't been initialized, return 1 to prevent detach and remove.
 
 // Return the number of references for the given volume
-func (d *VolumeDriver) getRefCount(vol string) uint {
+func (d *VolumeDriver) getRefCount(vol string) (uint, error) {
 	if d.refCounts.GetInitSuccess() != true {
-		return 1
+		return 0, fmt.Errorf(plugin_utils.ErrorPluginInit)
 	}
-	return d.refCounts.GetCount(vol)
+	return d.refCounts.GetCount(vol), nil
 }
 
 // Increment the reference count for the given volume
-func (d *VolumeDriver) incrRefCount(vol string) uint {
+func (d *VolumeDriver) incrRefCount(vol string) (uint, error) {
 	if d.refCounts.GetInitSuccess() != true {
-		return 1
+		return 0, fmt.Errorf(plugin_utils.ErrorPluginInit)
 	}
-	return d.refCounts.Incr(vol)
+	return d.refCounts.Incr(vol), nil
 }
 
 // Decrement the reference count for the given volume
 func (d *VolumeDriver) decrRefCount(vol string) (uint, error) {
 	if d.refCounts.GetInitSuccess() != true {
-		return 1, nil
+		return 0, fmt.Errorf(plugin_utils.ErrorPluginInit)
 	}
 	return d.refCounts.Decr(vol)
```

Comment:
I think the reason was that incr cannot yield an error : it will be just increment the count.
Decr however needs to indicate error if caller tried to decrement the count below zero.

For this PR, keeping the fix as small as possible and capturing the error in Remove handler only.